### PR TITLE
Fixes #2643 - Switch SslContextFactory.keystoreType from JKS to PKCS12.

### LINKED ISSUE #2643: Switch SslContextFactory.keystoreType from `JKS` to `PKCS12`
Since the default implementation in Java is now `PKCS12`, perhaps with Jetty 10 we should switch our defaults over to `PKCS12` as well, retiring the `JKS` default in SslContextFactory.

### PR DESCRIPTION
**Pull request by sbordet**
Updated old keystores to new PKCS12 keystores.
Removed unused keystores and truststores.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

Fixes #2643

### CODE REVIEW

File: jetty-alpn/jetty-alpn-conscrypt-server/src/test/java/org/eclipse/jetty/alpn/conscrypt/server/ConscryptHTTP2ServerTest.java [hunk 1]
```diff
@@ -79,11 +79,9 @@
     private void configureSslContextFactory(SslContextFactory sslContextFactory)
     {
         Path path = Paths.get("src", "test", "resources");
-        File keys = path.resolve("keystore").toFile();
+        File keys = path.resolve("keystore.p12").toFile();
         sslContextFactory.setKeyStorePath(keys.getAbsolutePath());
-        sslContextFactory.setKeyManagerPassword("OBF:1vny1zlo1x8e1vnw1vn61x8g1zlu1vn4");
-        sslContextFactory.setTrustStorePath(keys.getAbsolutePath());
-        sslContextFactory.setTrustStorePassword("OBF:1vny1zlo1x8e1vnw1vn61x8g1zlu1vn4");
+        sslContextFactory.setKeyStorePassword("OBF:1vny1zlo1x8e1vnw1vn61x8g1zlu1vn4");
```
Comment (gregw) [inline jetty-alpn/jetty-alpn-conscrypt-server/src/test/java/org/eclipse/jetty/alpn/conscrypt/server/ConscryptHTTP2ServerTest.java]: shall we not use obfuscation at all?

File: jetty-osgi/test-jetty-osgi/src/test/java/org/eclipse/jetty/osgi/test/TestJettyOSGiBootHTTP2Conscrypt.java [hunk 2]
```diff
@@ -135,13 +135,13 @@ public void testHTTP2() throws Exception
...
             Path path = Paths.get("src", "test", "config");
-            File keys = path.resolve("etc").resolve("keystore").toFile();
+            File keys = path.resolve("etc").resolve("keystore.p12").toFile();
 
             ClientConnector clientConnector = new ClientConnector();
             SslContextFactory.Client sslContextFactory = new SslContextFactory.Client();
-            sslContextFactory.setKeyManagerPassword("OBF:1vny1zlo1x8e1vnw1vn61x8g1zlu1vn4");
-            sslContextFactory.setTrustStorePath(keys.getAbsolutePath());
             sslContextFactory.setKeyStorePath(keys.getAbsolutePath());
+            sslContextFactory.setKeyStorePassword("OBF:1vny1zlo1x8e1vnw1vn61x8g1zlu1vn4");
```
Comment (gregw) [inline jetty-osgi/test-jetty-osgi/src/test/java/org/eclipse/jetty/osgi/test/TestJettyOSGiBootHTTP2Conscrypt.java]: no obfuscation

File: jetty-server/src/main/config/modules/test-keystore.mod [hunk 3]
```diff
@@ -9,13 +9,10 @@ ssl
 ssl
 
 [files]
-basehome:modules/test-keystore/keystore|etc/test-keystore
+basehome:modules/test-keystore/keystore.p12|etc/test-keystore.p12
```
Comment (gregw) [inline jetty-server/src/main/config/modules/test-keystore.mod]: can we just call the file either keystore.p12 or test-keystore.p12 and not rename during the move?

File: tests/test-integration/src/test/java/org/eclipse/jetty/test/HttpInputIntegrationTest.java [hunk 4]
```diff
@@ -104,9 +104,8 @@ public static void beforeClass() throws Exception
         // SSL Context Factory for HTTPS and HTTP/2
         String jettyDistro = System.getProperty("jetty.distro", "../../jetty-distribution/target/distribution");
         __sslContextFactory = new SslContextFactory.Server();
-        __sslContextFactory.setKeyStorePath(jettyDistro + "/../../../jetty-server/src/test/config/etc/keystore");
-        __sslContextFactory.setKeyStorePassword("OBF:1vny1zlo1x8e1vnw1vn61x8g1zlu1vn4");
-        __sslContextFactory.setKeyManagerPassword("OBF:1u2u1wml1z7s1z7a1wnl1u2g");
+        __sslContextFactory.setKeyStorePath(jettyDistro + "/../../../jetty-server/src/test/resources/keystore.p12");
```
Comment (gregw) [inline tests/test-integration/src/test/java/org/eclipse/jetty/test/HttpInputIntegrationTest.java]: can we lose the `../../..`?

**Review by gregw — APPROVED** [review]
I few quibbles, but fine eitherway.

File: jetty-alpn/jetty-alpn-conscrypt-server/src/test/java/org/eclipse/jetty/alpn/conscrypt/server/ConscryptHTTP2ServerTest.java (hunk 1 above)
Comment (sbordet) [reply jetty-alpn/jetty-alpn-conscrypt-server/src/test/java/org/eclipse/jetty/alpn/conscrypt/server/ConscryptHTTP2ServerTest.java]: No, it's just a test.

File: jetty-server/src/main/config/modules/test-keystore.mod (hunk 3 above)
Comment (sbordet) [reply jetty-server/src/main/config/modules/test-keystore.mod]: Okay.

File: tests/test-integration/src/test/java/org/eclipse/jetty/test/HttpInputIntegrationTest.java (hunk 4 above)
Comment (sbordet) [reply tests/test-integration/src/test/java/org/eclipse/jetty/test/HttpInputIntegrationTest.java]: Yes.

File: jetty-alpn/jetty-alpn-conscrypt-server/src/test/java/org/eclipse/jetty/alpn/conscrypt/server/ConscryptHTTP2ServerTest.java (hunk 1 above)
Comment (gregw) [reply jetty-alpn/jetty-alpn-conscrypt-server/src/test/java/org/eclipse/jetty/alpn/conscrypt/server/ConscryptHTTP2ServerTest.java]: but we use the password directly sometimes and obfuscate others.   We should just do one or the other everywhere.... but not really important.

Comment (sbordet) [reply jetty-alpn/jetty-alpn-conscrypt-server/src/test/java/org/eclipse/jetty/alpn/conscrypt/server/ConscryptHTTP2ServerTest.java]: I updated the tests to never use obfuscated passwords, which are now only used in specific tests and in configuration files (e.g. `realm.properties`, `*.mod` files, etc.).

**Pull request merged**